File: apps/android/app/src/main/java/dev/flextranslate/foundation/OfflineFirstState.kt
package dev.flextranslate.foundation

sealed interface OfflineFirstState {
    data object ReadyOfflineAsr : OfflineFirstState
    data class MissingOfflinePack(val packId: String) : OfflineFirstState
    data class UnsupportedOfflineTranslation(val languagePair: String, val deviceTier: String) : OfflineFirstState
    data object CloudDisabled : OfflineFirstState
    data class CaptureBlocked(val reason: String) : OfflineFirstState
}

File: apps/ios/FlexTranslate/Sources/FlexTranslate/OfflineFirstState.swift
import Foundation

enum OfflineFirstState: Equatable, CustomStringConvertible {
    case readyOfflineAsr
    case missingOfflinePack(packId: String)
    case unsupportedOfflineTranslation(languagePair: String, deviceTier: String)
    case cloudDisabled
    case captureBlocked(reason: String)

    var description: String {
        switch self {
        case .readyOfflineAsr: return "readyOfflineAsr"
        case let .missingOfflinePack(packId): return "missingOfflinePack(\(packId))"
        case let .unsupportedOfflineTranslation(languagePair, deviceTier): return "unsupportedOfflineTranslation(\(languagePair), \(deviceTier))"
        case .cloudDisabled: return "cloudDisabled"
        case let .captureBlocked(reason): return "captureBlocked(\(reason))"
        }
    }
}
